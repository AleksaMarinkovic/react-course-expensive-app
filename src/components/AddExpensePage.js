import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return  (
      <div>
        <div className="page-header">
          <div className="content-container">
            <div className="page-header__container">
              <h1 className="page-header__title">Add Expense</h1>
              <Link className="page-header__link" to="/dashboard">&lt;&lt; Back to my expenses</Link>
            </div>            
          </div>         
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>        
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);