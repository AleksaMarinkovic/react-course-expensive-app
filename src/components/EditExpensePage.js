import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startRemoveExpense, startEditExpense} from '../actions/expenses';
import { Link } from 'react-router-dom';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  onClick = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }
  render(){
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <div className="page-header__container">
              <h1 className="page-header__title">Editing expense</h1>
              <Link className="page-header__link" to="/dashboard">&lt;&lt; Back to my expenses</Link>
            </div>            
          </div>         
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button-secondary" onClick={this.onClick}>Remove</button>
        </div>        
      </div>
    );
  }
}
const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)  
});

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
}); 

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
