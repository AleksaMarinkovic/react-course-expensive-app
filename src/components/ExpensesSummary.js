import React from "react";
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from "numeral";

export const ExpensesSummary = ({expensesLength, total}) => 
{   
    const expenseWord = expensesLength === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(total / 100).format('$ 0,0.00');
    return (
        <div>
            {
                expensesLength > 0 && (<p>Currently showing {expensesLength} {expenseWord} totalling {formattedTotal}</p>)
            }
        </div>   
    );
};


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesLength: visibleExpenses.length,
        total: expensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);
