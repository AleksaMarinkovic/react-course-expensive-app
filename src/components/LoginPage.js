import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import GoogleButton from 'react-google-button';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>Get your expenses under control.</p>
            <hr className="box-layout__separator"/>
            <GoogleButton className="login-button" onClick={startLogin}>login</GoogleButton>
        </div>        
    </div>
);


const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);