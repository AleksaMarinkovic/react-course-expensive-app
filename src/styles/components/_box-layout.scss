.box-layout {
    background-image: url('/images/bg.jpg');
    background-color: lightblue;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100vw;
    position: fixed
}

.box-layout__box {
    background: fade-out($color: white, $amount: .05);
    border-radius: 5px;
    text-align: center;
    width: 25rem;
    padding: $l-size $m-size;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}

.box-layout__separator {
    border-top: 1px solid;
}