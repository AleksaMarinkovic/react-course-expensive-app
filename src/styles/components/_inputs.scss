.text-input {
    border: 1px solid #cacccd;
    height: 48px;
    padding: 10px;
    font-size: $font-size-large;
    font-weight: 300;
    border-radius: 2px;
}

.select {
    @extend .text-input;
    position: relative;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}