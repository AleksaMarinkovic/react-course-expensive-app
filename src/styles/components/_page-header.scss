.page-header {
    background: $off-white;
    padding: $l-size 0;
    margin-bottom: $l-size;
}

.page-header__title {
    margin: 0;
    font-weight: 300;
    span {
        font-weight: bold;
    }
}

.page-header__actions {
    margin-top: $m-size;
}

.page-header__container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: $s-size 0;
}

.page-header__link {
    margin: 0;
    margin-left: 4rem;
    text-decoration: none;
    font-weight: bold;
}