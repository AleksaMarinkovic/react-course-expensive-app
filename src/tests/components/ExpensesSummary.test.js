import shallow from "enzyme/build/shallow";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import React from 'react';

test('should matchsnapshot when 1 expense is showing', () => {
    const wrapper = shallow(<ExpensesSummary expensesLength={1} total={9500}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should matchsnapshot when 2 expenses are showing', () => {
    const wrapper = shallow(<ExpensesSummary expensesLength={2} total={75333}/>);
    expect(wrapper).toMatchSnapshot();
});