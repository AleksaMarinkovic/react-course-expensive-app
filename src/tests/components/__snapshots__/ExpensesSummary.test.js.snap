// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matchsnapshot when 1 expense is showing 1`] = `
<div>
  <p>
    Currently showing 
    1
     
    expense
     totalling 
    $ 95.00
  </p>
</div>
`;

exports[`should matchsnapshot when 2 expenses are showing 1`] = `
<div>
  <p>
    Currently showing 
    2
     
    expenses
     totalling 
    $ 753.33
  </p>
</div>
`;
